circuits/passportVerification/passportVerificationBuilder.circom

```diff
@@ -1,21 +1,126 @@
 pragma circom  2.1.6;
 
+include "../signatureVerifier/signatureVerification.circom";
+include "./passportVerificationFlow.circom";
+include "../hasher/passportHash.circom";
+include "circomlib/circuits/bitify.circom";
+include "circomlib/circuits/poseidon.circom";
 
 template PassportVerificationBuilder(
-        DG1_SIZE,
-        DG15_SIZE,
-        ENCAPSULATED_CONTENT_SIZE,
-        SIGNED_ATTRIBUTES_SIZE,
-        HASH_BLOCK_SIZE,
-        HASH_TYPE,
-        SIGNATURE_TYPE,
-        CHUNK_SIZE,
-        CHUNK_NUMBER,
-        TREE_DEPTH,
-        AA_FLOWS_NUMBER,
-        AA_FLOWS_BITMASK,
-        NoAA_FLOWS_NUMBER,
-        NoAA_FLOWS_BITMASK
+    DG1_SIZE,
+    DG15_SIZE,
+    ENCAPSULATED_CONTENT_SIZE,
+    SIGNED_ATTRIBUTES_SIZE,
+    HASH_BLOCK_SIZE,
+    HASH_TYPE,
+    SIGNATURE_TYPE,
+    SALT_LEN,
+    E_BITS,
+    CHUNK_SIZE,
+    CHUNK_NUMBER,
+    TREE_DEPTH,
+    AA_FLOWS_NUMBER,
+    AA_FLOWS_BITMASK,
+    NoAA_FLOWS_NUMBER,
+    NoAA_FLOWS_BITMASK
 ) {
 
+    var DG1_LEN                  = DG1_SIZE                  * HASH_BLOCK_SIZE;
+    var DG15_LEN                 = DG15_SIZE                 * HASH_BLOCK_SIZE;
+    var ENCAPSULATED_CONTENT_LEN = ENCAPSULATED_CONTENT_SIZE * HASH_BLOCK_SIZE;
+    var SIGNED_ATTRIBUTES_LEN    = SIGNED_ATTRIBUTES_SIZE    * HASH_BLOCK_SIZE;
+
+    var PUBKEY_LEN;
+    var SIGNATURE_LEN;
+
+    //ECDSA
+    if (SIGNATURE_TYPE > 5){
+        PUBKEY_LEN    = 2 * CHUNK_NUMBER * CHUNK_SIZE;
+        SIGNATURE_LEN = 2 * CHUNK_NUMBER * CHUNK_SIZE;   
+    }
+    //RSA
+    if (SIGNATURE_TYPE <= 5){
+        PUBKEY_LEN    = CHUNK_NUMBER;
+        SIGNATURE_LEN = CHUNK_NUMBER;
+    }
+
+
+    signal input encapsulatedContent         [ENCAPSULATED_CONTENT_LEN];
+    signal input dg1                         [DG1_LEN                 ];
```

Comment:
Make [TEXT] format, instead of [TEXT      ]

Reply:
DOne